src/Common/src/System/Runtime/CompilerServices/__BlockAllReflectionAttribute.cs

```diff
@@ -0,0 +1,17 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+/*
+  Providing a definition for __BlockAllReflectionAttribute in an assembly is a signal to the .NET Native toolchain 
+  to remove the metadata for all APIs. This both reduces size and disables all reflection on those 
+  APIs in libraries that include this.
+*/
+
+using System;
+
+namespace System.Runtime.CompilerServices
+{
+    [AttributeUsage(AttributeTargets.All)]
+    internal class __BlockAllReflectionAttribute : Attribute { }
```

Comment:
Nit: The double underscores just make ugly names, and I do not think they buy us anything. Should we stop spreading them around?

(I know we have a prior art here with __BlockReflectionAttribute ... that one can be renamed too.)

Reply:
The entire scheme of "presence of the type activates the feature" is kind of ugly. `__BlockReflectionAttribute` cannot be easily renamed as that one is also hardcoded in CoreFX (where it annotates e.g. Microsoft.CSharp) and in the Project N toolchain. Using the same name here makes the blocking of things coming from CoreFX work for free.

I would rather we switch the former to the public [DisablePrivateReflectionAttribute](https://msdn.microsoft.com/en-us/library/system.runtime.compilerservices.disableprivatereflectionattribute(v=vs.110).aspx) and introduce a new *public* attribute for the latter.

Reply:
(I can try to push for this if you think it makes sense, but I'm not sure how successful I'll be.)

Reply:
Ok, this can wait for later. I agree that it would be nice to have public way to do this.